ldap/servers/slapd/back-ldbm/db-bdb/bdb_layer.c

```diff
@@ -2988,6 +2997,82 @@ perf_threadmain(void *param)
     return 0;
 }
 
+
+/*
+ * create a thread for locks_monitoring_threadmain
+ */
+static int
+bdb_start_locks_monitoring_thread(struct ldbminfo *li)
+{
+    int return_value = 0;
+    if ((li->li_dblock_monitoring) && (li->li_dblock_threshold)) {
```

Comment:
Could it happen that there is no dblock threshold set but dblock monitoring is enabled, if it did then the dblock monitoring thread wouldn't start.

Reply:
It is intentional. If the threshold is empty or 0, it won't start as we need both attributes for the monitoring to function.